### PR TITLE
vscode-extensions.fill-labs.dependi: init at 0.7.2

### PR DESCRIPTION
**Pull request by 21CSM**
## Description of changes

Adds [dependi](https://marketplace.visualstudio.com/items?itemName=fill-labs.dependi), which is a new extension that deprecates [crates](https://marketplace.visualstudio.com/items?itemName=serayuzgur.crates).

Also, add myself as a maintainer.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -1814,6 +1814,23 @@ let
...
+      fill-labs.dependi = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "dependi";
+          publisher = "fill-labs";
+          version = "0.7.1";
+          hash = "sha256-RCEh3cy4JVwdEEY3LVAj+0yDQ/6KgvL3QvQEbU5EfYw=";
+        };
+        meta = {
+          changelog = "https://marketplace.visualstudio.com/items/fill-labs.dependi/changelog";
+          description = "Empowers developers to efficiently manage dependencies and address vulnerabilities in Rust, Go, JavaScript, and Python projects.";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1826]:
```suggestion
          description = "VSCode extension for managing dependencies and address vulnerabilities in Rust, Go, JavaScript, and Python projects";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -1814,6 +1814,23 @@ let
         };
       };
 
+      fill-labs.dependi = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "dependi";
+          publisher = "fill-labs";
+          version = "0.7.1";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1821]:
```suggestion
          version = "0.7.2";
```
0.7.2 is available upstream

Comment (21CSM) [reply pkgs/applications/editors/vscode/extensions/default.nix:1821]: Thanks for the feedback. When I make this change, shall I amend the commit and PR title to 'unit at 7.1.2'?

Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/default.nix:1821]: Yes. Title will need update. And it should be a single commit.

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 1 above)
Comment (21CSM) [reply pkgs/applications/editors/vscode/extensions/default.nix]: Thanks for the feedback, addressed it.

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 2 above)
Comment (21CSM) [reply pkgs/applications/editors/vscode/extensions/default.nix]: Thanks, I have updated the PR. Let me know if anything needs to be modified.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -1814,6 +1814,23 @@ let
...
+      fill-labs.dependi = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "dependi";
+          publisher = "fill-labs";
+          version = "0.7.2";
+          hash = "sha256-S3R1oLk7facP5Rn9czmHlffhMtLNrSaGYbaU3/x6/aM=";
+        };
+        meta = {
+          changelog = "https://marketplace.visualstudio.com/items/fill-labs.dependi/changelog";
+          description = "VSCode extension for managing dependencies and address vulnerabilities in Rust, Go, JavaScript, and Python projects.";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1826]:
```suggestion
          description = "VSCode extension for managing dependencies and address vulnerabilities in Rust, Go, JavaScript, and Python projects";
```
It should not end with a dot (`.`).

Comment (21CSM) [reply pkgs/applications/editors/vscode/extensions/default.nix]: My bad, missed that in the last review comment. Addressed.

**Review by superherointj — APPROVED** [review]

**Pull request merged**